Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -636,6 +636,7 @@ static void AddGCAdapter(libusb_device* device)
       }
     }
   }
+  libusb_free_config_descriptor(config);
```

Comment:
If you can refactor this to `MakeConfigDescriptor()` then I think that's the better choice, just for the RAII cleanup.

Reply:
- I assume I'd write something like `const auto [error, config] = LibusbUtils::MakeConfigDescriptor(device)` to match the existing calls to this function.
	- I don't really like that you don't see the types of `error` and `config` anymore, but there's not really a better way to use `MakeConfigDescriptor()`.
- Do I need to introduce a new scope to eagerly destroy the `ConfigDescriptor` (unique_ptr with dynamic destructor) when not in use? If not, should I reset the unique_ptr or leave it initialized until the function returns? And should I reuse the `error` value as the return value of `libusb_interrupt_transfer()` below?